src/plugins/discord/mirror.test.js

```diff
@@ -0,0 +1,320 @@
+          authorId: "4",
+        });
+      }
+
+      const result = mirror.messages("2");
+      expect(result).toEqual([mes1, mes2]);
+    });
+  });
+
+  describe("reactions", () => {
+    it("inserts reactions", async () => {
+      const reaction = {
+        emoji: {id: "id", name: "name"},
+        channelId: "2",
+        messageId: "3",
+        authorId: "4",
+      };
+      const db = new Database(":memory:");
+      const mirror = new Mirror(db, "0");
+      mirror.addReaction(reaction);
+      const result = await db
+        .prepare(
+          dedent`\
+          SELECT
+            channel_id,
+            author_id,
+            message_id,
+            emoji
+          FROM message_reactions`
+        )
+        .get();
+      expect(result).toEqual({
+        emoji: "name:id",
+        channel_id: "2",
+        message_id: "3",
+        author_id: "4",
+      });
+    });
+    it("retrieves reactions", () => {
+      const reaction1 = {
+        emoji: {id: "id", name: "name"},
+        channelId: "1",
+        messageId: "2",
+        authorId: "3",
+      };
+
+      const reaction2 = {
+        emoji: {id: null, name: "emojiName"},
+        channelId: "1",
+        messageId: "2",
+        authorId: "4",
+      };
+
+      const db = new Database(":memory:");
+      const mirror = new Mirror(db, "0");
+      mirror.addReaction(reaction1);
+      mirror.addReaction(reaction2);
+      const result = mirror.reactions("1", "2");
+      expect(result).toEqual([reaction1, reaction2]);
+    });
+  });
+
+  describe("mentions", () => {
+    it("inserts mentions", async () => {
+      const db = new Database(":memory:");
+      const mirror = new Mirror(db, "0");
+      mirror.addMention(testMessage("1"), "45");
+      const result = await db
+        .prepare(
+          `\
+        SELECT
+          channel_id,
+          message_id,
+          user_id
+        FROM message_mentions`
+        )
+        .get();
+      expect(result).toEqual({channel_id: "2", message_id: "1", user_id: "45"});
+    });
+
+    it("retrieves mentions", () => {
+      const db = new Database(":memory:");
+      const mirror = new Mirror(db, "0");
+      mirror.addMention(testMessage("1"), "56");
+      mirror.addMention(testMessage("1"), "78");
+      const result = mirror.mentions("2", "1");
+      expect(result).toEqual(["56", "78"]);
+    });
+  });
+
+  describe("reactionEmoji", () => {
+    it("retrieves emojies", () => {
+      const reaction1 = {
+        emoji: {id: "id", name: "name"},
+        channelId: "1",
+        messageId: "2",
+        authorId: "3",
+      };
+
+      const reaction2 = {
+        emoji: {id: null, name: "emojiName"},
+        channelId: "1",
+        messageId: "2",
+        authorId: "4",
+      };
+
+      const db = new Database(":memory:");
+      const mirror = new Mirror(db, "0");
+      mirror.addReaction(reaction1);
+      mirror.addReaction(reaction2);
+      const result = mirror.reactionEmoji("1", "2");
```

Comment:
In cases like this "0", "1" and "2" are not very informative, I think it's better to use clear names.
So this style would be clearer:

```js
const guildId: Snowflake = "0";
const channelId: Snowflake = "1";
const messageId: Snowflake = "2";
const reaction1: Reaction = {
  // ...
  channelId,
  messageId,
};

// ...
const mirror = new Mirror(db, guildId);
// ...
mirror.reactionEmoji(channelId, messageId);
```